### PR TITLE
xpumanager sidecar: add note about using HTTPS with xpum

### PR DESCRIPTION
**Pull request by tkatila**

### CODE REVIEW

**Comment by tkatila** [comment]
fyi, this does not go through CI as it's only doc change.

**Review by mythi — APPROVED** [review]

File: cmd/xpumanager_sidecar/README.md [hunk 1]
```diff
@@ -64,11 +65,60 @@ $ kubectl patch daemonsets.apps intel-xpumanager --patch-file 'https://github.co
...
+
+### Use HTTPS with XPU-Manager
+
+XPU-Manager can be configured to use HTTPS on the metrics interface. For the gunicorn sidecar, cert and key files have to be added to the command:
+```
+      - command:
+        - gunicorn
+...
+        - --certfile=/certs/tls.crt
+        - --keyfile=/certs/tls.key
```
Comment (eero-t) [inline cmd/xpumanager_sidecar/README.md:85]: Add also example of mounting the secret volume(s) for these?

Comment (tkatila) [reply cmd/xpumanager_sidecar/README.md:85]: I added it in the cert-manager section below.

**Review by uniemimu — APPROVED** [review]

File: cmd/xpumanager_sidecar/README.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Table of Contents
   * [Install XPU-Manager with the Sidecar](#install-xpu-manager-with-the-sidecar)
   * [Install Sidecar to an Existing XPU-Manager](#install-sidecar-to-an-existing-xpu-manager)
 * [Verify Sidecar Functionality](#verify-sidecar-functionality)
+* [Use HTTPS with XPU-Manager]()
```
Comment (eero-t) [inline cmd/xpumanager_sidecar/README.md:11]: XPU Manager project writes the name everywhere without '-': https://github.com/intel/xpumanager

so I think this doc should also write it like that.

And this entry should also have link to the corresponding section.

File: cmd/xpumanager_sidecar/README.md [hunk 3]
```diff
@@ -64,11 +65,73 @@ $ kubectl patch daemonsets.apps intel-xpumanager --patch-file 'https://github.co
...
+spec:
+  dnsNames:
+  - xpum.svc
+  - xpum.svc.cluster.local
+  issuerRef:
+    kind: Issuer
+    name: selfsigned-issuer
+  secretName: xpum-server-cert
+```
+One would then include `xpum-server-cert` in the XPU-manager deployment.
```
Comment (eero-t) [inline cmd/xpumanager_sidecar/README.md:116]: As this part is not specific to using `cert-manager` for generating the secrets, I do not think it should be hidden under (same) `<details>`.

IMHO better would be to have it earlier, maybe in same example with `gunicorn` args:
```
    volumes:
    - name: certs
      secret:
        defaultMode: 420
        secretName: xpum-server-cert
...
    containers:
      - name: python-exporter
        volumeMounts:
        - mountPath: /certs
          name: certs
          readOnly: true
...
      - command:
        - gunicorn
...
        - --certfile=/certs/tls.crt
        - --keyfile=/certs/tls.key
...
        - xpum_rest_main:main()
```

**Review by eero-t — CHANGES_REQUESTED** [review]

File: cmd/xpumanager_sidecar/README.md [hunk 4]
```diff
@@ -22,13 +23,13 @@ Intel GPUs can be interconnected via an XeLink. In some workloads it is benefici
 | -startup-delay | int | 10 | Startup delay before the first topology fetching (seconds, >= 0) |
 | -label-namespace | string | gpu.intel.com | Namespace or prefix for the labels. i.e. **gpu.intel.com**/xe-links |
 | -allow-subdeviceless-links | bool | false | Include xelinks that are not on subdevices |
```
Comment (eero-t) [inline cmd/xpumanager_sidecar/README.md:25]: Name of the option indicates that the description should rather be:
```suggestion
| -allow-subdeviceless-links | bool | false | Include xelinks also for devices that do not have subdevices |
```

**Review by eero-t — DISMISSED** [review]
Approved, but option description (not touched here) seems like it should be fixed too?

**Pull request merged**